yogstation/code/modules/client/verbs/antag_token.dm

```diff
@@ -0,0 +1,92 @@
+//2.5 minutes, gotta prevent DDoSing using SQL
+#define AntagTokenCooldown 1500
+
+//List of all people using antag tokens this round
+GLOBAL_LIST_EMPTY(antag_token_users)
+
+/client/verb/antag_token_check()
+	set name = "Check/Use Antag Token"
+	set category = "OOC"
+	set desc = "Check if you have an antag token, and if you do, use it."
+	var/client/C = usr.client
+
+	if(SSticker.current_state > GAME_STATE_PREGAME)
+		to_chat(usr, "<span class='userdanger'>You must use this in the lobby!</span>")
+		return
+
+	if(world.time < C.last_antag_token_check)
+		to_chat(usr, "<span class='userdanger'>You cannot use this verb yet! Please wait.</span>")
+		return
+		
+	var/datum/DBQuery/query_antag_token_existing = SSdbcore.NewQuery({"SELECT ckey FROM [format_table_name("antag_tokens")] WHERE ckey = '[sanitizeSQL(ckey(ckey))]' AND redeemed = 0"})
+
+	if(!query_antag_token_existing.warn_execute())
+		qdel(query_antag_token_existing)
+		return
+
+	C.last_antag_token_check = world.time + AntagTokenCooldown
+
+	if(query_antag_token_existing.NextRow())
+		if(alert("You have an antag token! Do you want to use it? YOU CAN GET ANTAG'S THAT YOU HAVE DISABLED IN YOUR PREFERENCES",, "Yes", "No") != "Yes")
```

Comment:
This is not true, the way you have it setup, you can be only an antag you enabled in your prefs. Which sounds problematic as someone could just try each time but with only like, wizard enabled so that he only gets his thing on a wizard round

Reply:
We pick from the list of all users attempting to use an antag token, we don't get_players_for_role() for a specific role